src/Microsoft.ML.AutoML/TrainerExtensions/SweepableParams.cs

```diff
@@ -124,7 +124,7 @@ public static IEnumerable<SweepableParam> BuildMatrixFactorizationParams()
                 new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.LearningRate), new object[] { 0.001f, 0.01f, 0.1f }),
                 new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.ApproximationRank), new object[] { 8, 16, 64, 128 }),
                 new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.Lambda), new object[] { 0.01f, 0.05f, 0.1f, 0.5f, 1f }),
-                new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.LossFunction), new object[] { MatrixFactorizationTrainer.LossFunctionType.SquareLossRegression, MatrixFactorizationTrainer.LossFunctionType.SquareLossOneClass }),
+                new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.LossFunction), new object[] { "MatrixFactorizationTrainer.LossFunctionType.SquareLossRegression", "MatrixFactorizationTrainer.LossFunctionType.SquareLossOneClass" }),
```

Comment:
> @LittleLittleCloud [Dec 16, 2019, 3:28 PM PST](https://github.com/dotnet/machinelearning/issues/4579#issuecomment-566296343)
>
> -----
> The root cause for that is in CodeGenerator, it only handle cases when trainer option's value type is built-in type. see the following code link.
> https://github.com/dotnet/machinelearning/blob/f1f8942a8272a9c87373d11bc89467461c8ecad1/src/Microsoft.ML.CodeGenerator/CodeGenerator/CSharp/TrainerGeneratorBase.cs#L56
> 
> While in this case, the null-value field's type is a `MatrixFactorizationTrainer.LossFunctionType`. So it's been ignored and assigned a null value.
> 
> A quick fix can be fairly simple: just pass the string value of `MatrixFactorizationTrainer.LossFunctionType` in sweepable parameters. 
> Or we can use reflection to get the right type.

Will changing this to a string affect our ability to sweep over the values? I think we need the ability pass the native objects for sweeping.

Reply:
Oh right it should fail. But when testing it runs pretty well... Now I'm even more confusing. I will dig into more...